tests/integration/test_task_pool.py

```diff
@@ -1243,6 +1245,145 @@ async def test_detect_incomplete_tasks(
             # ensure that it is correctly identified as incomplete
             assert itask.state.outputs.get_incomplete()
             assert itask.state.outputs.is_incomplete()
-            assert log_filter(log, contains=f"[{itask}] did not complete required outputs:")
+            assert log_filter(
+                log, contains=f"[{itask}] did not complete required outputs:")
             # the task should not have been removed
             assert itask in schd.pool.get_tasks()
+
+
+@pytest.mark.parametrize('compat_mode', ['compat-mode', 'normal-mode'])
+@pytest.mark.parametrize('cycling_mode', ['integer', 'datetime'])
+async def test_compute_runahead(
+    cycling_mode,
+    compat_mode,
+    flow,
+    scheduler,
+    start,
+    monkeypatch,
+):
+    """Test the calculation of the runahead limit.
+
+    This test ensures that:
+    * Runahead tasks are excluded from computations
+      see https://github.com/cylc/cylc-flow/issues/5825
+    * Tasks are initiated with the correct is_runahead status on statup.
+    * Behaviour in compat/regular modes is same unless failed tasks are present
+    * Behaviour is the same for integer/datetime cycling modes.
+
+    """
+    if cycling_mode == 'integer':
+        config = {
+            'scheduler': {
+                'allow implicit tasks': 'True',
+            },
+            'scheduling': {
+                'initial cycle point': '1',
+                'cycling mode': 'integer',
+                'runahead limit': 'P3',
+                'graph': {
+                    'P1': 'a'
+                },
+            }
+        }
+        point = lambda point: IntegerPoint(str(int(point)))
+    else:
+        config = {
+            'scheduler': {
+                'allow implicit tasks': 'True',
+                'cycle point format': 'CCYY',
+                'cycle point time zone': 'Z'
```

Comment:
Without `cycle point time zone = Z`, the second parameterized test fails (in my time zone at least) with:
```
FAILED tests/integration/test_task_pool.py::test_compute_runahead[datetime-normal-mode] - 
   cylc.flow.exceptions.SequenceDegenerateError:
      R/0001/P1Y, point format CCYY: equal adjacent points: 0001 => 000
```

but it passes if I run the exact same test individually 🤯

Reply:
This is because in back-compat mode the cycle point time zone is assumed to be local, whereas in normal mode it is assumed to be UTC, and this was contaminating the point parsing cache between the tests. I've opened a PR against this one that ensures the cycle point time zone is taken into account when accessing the cache: https://github.com/hjoliver/cylc-flow/pull/37

Reply:
Brilliant, thanks.